src/utils/price.ts

```diff
@@ -0,0 +1,77 @@
+import BN from 'bn.js'
+import { assert } from './miscellaneous'
+import { TEN, UNLIMITED_ORDER_AMOUNT_BIGNUMBER } from 'const'
+import BigNumber from 'bignumber.js'
+
+export function adjustAmountToLowerPrecision(amount: BN, higherPrecision: number, lowerPrecision: number): BN {
+  if (higherPrecision === lowerPrecision) {
+    // no adjustment required
+    return amount
+  }
+
+  assert(higherPrecision > lowerPrecision, 'higherPrecision must be > lowerPrecision')
```

Comment:
If we have this assert, means this util need to be called with the precisions in the right order. Wouldn't it be better to allow this util to provide the two precisions without any assumptions?

We would check which one is bigger inside the util (once) instead of doing on every call to the util.

Reply:
> If we have this assert, means this util need to be called with the precisions in the right order. Wouldn't it be better to allow this util to provide the two precisions without any assumptions?

That is something I considered afterwards. Instead of the assert, the function would select the higher precision and adjust from there.
My concern with this approach is that this can lead to a silent failure.

Let's say we have token `A` precision 12, token `B` precision 2.
If due to some bug we selected `B` with the higher precision and called the function with (amount, 2, 12).
In the current implementation, it throws and error.
In the new approach, it simply adjusts the precision assuming 12 is the higher amount.

Another way of looking at this is that it isn't this function responsibility to deal with bug somewhere else, and the function should do what it's supposed to do: adjust to lower precision.

/Conflicted.

> We would check which one is bigger inside the util (once) instead of doing on every call to the util.

I don't get that. For every precision we need to either do the assert or check which is bigger, so every call still needs to do at least one check. No?

Reply:
Understood, it's fine then